app/scripts/modules/core/src/pipeline/config/triggers/Trigger.tsx

```diff
@@ -0,0 +1,209 @@
+    if (expectedArtifacts) {
+      const editingArtifact = expectedArtifacts.findIndex(artifact => artifact.id === expectedArtifact.id);
+      if (editingArtifact >= 0) {
+        const newExpectedArtifactsList = expectedArtifacts.splice(0);
+        newExpectedArtifactsList.splice(editingArtifact, 1, expectedArtifact);
+        this.props.updateExpectedArtifacts(newExpectedArtifactsList);
+      } else {
+        this.props.updateExpectedArtifacts([...expectedArtifacts, expectedArtifact]);
+      }
+    } else {
+      this.props.updateExpectedArtifacts([expectedArtifact]);
+    }
+
+    const expectedArtifactIds = this.props.trigger.expectedArtifactIds;
+    if (expectedArtifactIds && !expectedArtifactIds.includes(expectedArtifact.id)) {
+      this.triggerUpdated({
+        expectedArtifactIds: [...expectedArtifactIds, expectedArtifact.id],
+      });
+    } else {
+      this.triggerUpdated({
+        expectedArtifactIds: [expectedArtifact.id],
+      });
+    }
+  };
+
+  private changeExpectedArtifacts = (expectedArtifacts: string[]) => {
+    this.triggerUpdated({
+      expectedArtifactIds: expectedArtifacts,
+    });
+  };
+
+  public render() {
+    const { triggerConfig, type } = this.state;
+    const { pipeline, trigger } = this.props;
+    const { TriggerContents } = this;
+    return (
+      <div className="row">
+        <div className="col-md-12">
+          <div className="form-horizontal panel-pipeline-phase">
+            <div className="form-group row">
+              <div className="col-md-10">
+                <div className="row">
+                  <label className="col-md-3 sm-label-right">Type</label>
+                  <div className="col-md-4">
+                    <Select
+                      className="input-sm"
+                      clearable={false}
+                      onChange={this.handleTypeChange}
+                      options={[
+                        { label: 'Select...', value: '' },
+                        ...Registry.pipeline.getTriggerTypes().map(t => ({
+                          label: t.label,
+                          value: t.key,
+                        })),
+                      ]}
+                      value={type}
+                    />
+                  </div>
+                  <div className="col-md-5">{triggerConfig && triggerConfig.description}</div>
+                </div>
+              </div>
+              <div className="col-md-2 text-right">
+                <button className="btn btn-sm btn-default" onClick={this.removeTrigger}>
+                  <span className="glyphicon glyphicon-trash" />
+                  <span className="visible-xl-inline">Remove trigger</span>
+                </button>
+              </div>
+            </div>
+            <div className="form-group">
+              <div className="col-md-10">
+                <TriggerContents />
+              </div>
+            </div>
+            <div className="form-group">
```

Comment:
This should be conditional on enabled feature.  

The angular code wraps it in ```<render-if-feature feature="artifactsRewrite">```

See my other comment in the AngularJS code, I think there may be another missing condition, when `rewriteArtifacts` is enabled.

Reply:
please see previous comments: https://github.com/spinnaker/deck/pull/7046#discussion_r291660734